### PR TITLE
Updated to-test.md for 12.0-beta.

### PR DESCRIPTION
**Pull request by zinigor**
This PR fixes the to-test.md header and removes testing instructions that can't be carried out by the general audience.

## Proposed changes:
* Changed to-test.md.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Open to-test.md and observe the header that says 12.0 instead of 11.9.

### CODE REVIEW

File: projects/plugins/jetpack/to-test.md [hunk 1]
```diff
@@ -59,14 +52,6 @@ This version of Jetpack included several small fixes to ensure it’s compatible
...
-
-**Use a WoA site.**
-
-- Open the wp-admin of the site on mobile (or simulate the mobile in your desktop browser)
-- Click the WordPress logo in the left-hand top corner to open the unified navigation
-- Click the WordPress logo again, and make sure the navigation closes.
-
 ### And More!
 
 You can see a [full list of changes in this release here](https://github.com/Automattic/jetpack/blob/jetpack/branch-11.9/projects/plugins/jetpack/CHANGELOG.md). Please feel free to test any and all functionality mentioned!
```
Comment (samiff) [inline projects/plugins/jetpack/to-test.md]: This link needs to be updated.

File: projects/plugins/jetpack/to-test.md [hunk 2]
```diff
@@ -29,13 +29,6 @@ No new features were added, so the only thing to test here is that they work as
...
-### Golden ticket
-
-- Go to the golden-token MC tool https://mc.a8c.com/jetpack/golden-token/
-- Generate a new license key and assign it to your test site using URL or the Blog ID
-- Go to /wp-admin/admin.php?page=jetpack#/my-plan
-- Verify that the Jetpack Golden Token plan appears
-
 ### WordPress 6.2 compatibility
 
 This version of Jetpack included several small fixes to ensure it’s compatible with the latest WordPress. Most test sites are already set to use the WordPress 6.2 RC version. However, if you are using Jurassic Ninja, Atomic, or your standalone site, you may need to install the WordPress beta tester plugin, or you have set it up to use the appropriate version in some other way (through the Pressable dashboard). Also, by the time you try testing, the WP 6.2 may already be published.
```
Comment (samiff) [inline projects/plugins/jetpack/to-test.md]: > Most test sites are already set to use the WordPress 6.2 RC version. However, if you are using Jurassic Ninja, Atomic, or your standalone site, you may need to install the WordPress beta tester plugin

This information doesn't seem tailored for general audience.

File: projects/plugins/jetpack/to-test.md (hunk 1 above)
Comment (zinigor) [reply projects/plugins/jetpack/to-test.md]: Updated, thank you!

File: projects/plugins/jetpack/to-test.md (hunk 2 above)
Comment (zinigor) [reply projects/plugins/jetpack/to-test.md]: Yes, you're right, I have changed the paragraph to mention that 6.2 is already out.

**Review by samiff — APPROVED** [review]

**Pull request merged**